### PR TITLE
feat: Updated Workflows, and Cron created Workflows, to ensure they get their own pod by using Pod AntiAffinity. This replaces resource limits and/or requests.

### PR DESCRIPTION
**Pull request by aleksandrmelnikov**
**What this PR does**:
This PR will ensure that only one workflow will be scheduled on a Node.
This way, that workflow will get the entire Node resources to execute.
- All the templates used by the workflow will get a dedicated Node. So multiple containers will get their own Node.

**Which issue(s) this PR fixes**:

Fixes onepanelio/core#583

**Special notes for your reviewer**:
The changes here may not exclude the situation where a Workspace shares a node with a Workflow.
Should I adjust the code to ensure a workspace does not share a node with a workflow?

### CODE REVIEW

**Comment by rushtehrani** [comment]
@aleksandrmelnikov yes, we need to make sure a Workspace doesn't share a node with the Workflow, otherwise there will be an issue with resource allocation.  For example, the Workspace and Workflow might both try to request a GPU.

File: pkg/workflow_execution.go [hunk 1]
```diff
@@ -401,6 +361,49 @@ func (c *Client) createWorkflow(namespace string, workflowTemplateID uint64, wor
...
+			break
+		}
+		template.Metadata.Labels = map[string]string{antiAffinityLabelKey: nodeSelectorVal}
+		addPodAffinity = true
+	}
+	if addPodAffinity {
+		wf.Spec.Affinity = &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+					{LabelSelector: &metav1.LabelSelector{
```
Comment (rushtehrani) [inline pkg/workflow_execution.go:394]: @aleksandrmelnikov can you clean this up and move it to next line?

File: pkg/workflow_execution.go [hunk 2]
```diff
@@ -401,6 +361,49 @@ func (c *Client) createWorkflow(namespace string, workflowTemplateID uint64, wor
...
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{Key: antiAffinityLabelKey, Operator: "In", Values: []string{nodeSelectorVal}},
+						},
+					},
+						TopologyKey: "kubernetes.io/hostname"},
```
Comment (rushtehrani) [inline pkg/workflow_execution.go:399]: This also seems to be indented incorrectly. I think `go fmt` should've caught this?

File: pkg/workflow_execution.go (hunk 1 above)
Comment (aleksandrmelnikov) [reply pkg/workflow_execution.go:394]: Yes, will do

Comment (aleksandrmelnikov) [reply pkg/workflow_execution.go]: Check now?

File: pkg/workflow_execution.go (hunk 2 above)
Comment (aleksandrmelnikov) [reply pkg/workflow_execution.go]: Check now?
I've had to move it manually on it's own line for go fmt to do something about it.

Comment (Vafilor) [reply pkg/workflow_execution.go]: @rushtehrani Surprisingly it doesn't. I also tried it and get the exact same result.

But if you change some whitespace Like having `key` in  `{Key:...` go on the next line, it formats it differently.

**Pull request merged**